aea/cli/remove.py

```diff
@@ -87,65 +105,330 @@ def skill(ctx: Context, skill_id):
+        self.item_name = item_id.name
+
+        self.current_item = self.get_current_item()
+        self.required_by: Set[PackageId] = set()
+        self.dependencies_can_be_removed: Set[PackageId] = set()
+        try:
+            (
+                self.required_by,
+                self.dependencies_can_be_removed,
+                *_,
+            ) = ItemRemoveHelper(self.agent_config).check_remove(
+                self.item_type, self.current_item
+            )
+        except FileNotFoundError:
+            pass  # item registered but not present on filesystem
+
+    def get_current_item(self) -> PublicId:
+        """Return public id of the item already presents in agent config."""
+        current_item = get_item_public_id_by_author_name(
+            self.ctx.agent_config,
+            self.item_type,
+            self.item_id.author,
+            self.item_id.name,
+        )
+        if not current_item:  # pragma: nocover # actually checked in check_item_present
             raise click.ClickException(
-                "{} {} not found. Aborting.".format(item_type.title(), item_name)
+                "The {} '{}' is not supported.".format(self.item_type, self.item_id)
             )
-        if (
-            item_folder.exists() and not ctx.agent_config.author == item_id.author
-        ):  # pragma: no cover
+        return current_item
+
+    def remove(self) -> None:
+        """Remove item and it's dependencies if specified."""
+        click.echo(
+            "Removing {item_type} '{item_name}' from the agent '{agent_name}'...".format(
+                agent_name=self.agent_name,
+                item_type=self.item_type,
+                item_name=self.item_name,
+            )
+        )
+        self.remove_item()
+        if self.with_dependencies:
+            self.remove_dependencies()
+        click.echo(
+            "{item_type} '{item_name}' was removed from the agent '{agent_name}'...".format(
+                agent_name=self.agent_name,
+                item_type=self.item_type,
+                item_name=self.item_name,
+            )
+        )
+
+    @property
+    def agent_items(self) -> Set[PublicId]:
+        """Return items registered with agent of the same type as item."""
+        return getattr(self.agent_config, self.item_type_plural, set)
+
+    @property
+    def is_required_by(self) -> bool:
+        """Is required by any other registered component in the agent."""
+        return bool(self.required_by)
+
+    def remove_item(self) -> None:
+        """
+        Remove item.
+
+        Removed from the filesystem.
+        Removed from the agent configuration
+
+        Does not remove dependencies, please use `remove_dependencies`.
+        """
+        if (not self.force) and self.is_required_by:
+            raise click.ClickException(
+                f"Package {self.item_type} {self.item_id} can not be removed cause required by {','.join(map(str, self.required_by))}"
```

Comment:
```suggestion
                f"Package {self.item_type} {self.item_id} can not be removed because it is required by {','.join(map(str, self.required_by))}"
```

Reply:
done